kong/tools/public.lua

```diff
@@ -164,6 +165,37 @@ do
 
     return args, err_code, raw_body, req_mime
   end
+
+
+  -- Obtain the unique node id for this node.
+  do
+    local node_id
+    function _M.get_node_id()
```

Comment:
I'm a bit worried that we will see this utility being used in the `init` phase soon - and since we do nothing to guard against it, the issue would silently go unnoticed. This sort of issue has happened _so many times_ in this project that it is one of the reasons behind my suggestion to put this in a controlled API if it is to be consumed publicly.

Until the proper API (with `init` and `init_worker` seeding and the proper initialization mechanisms are in place) - can we make this function throw an error if called in `init`?
```lua
if ngx.get_phase() == "init" then
  error("API disabled in the context of init_by_lua", 2)
end
```

Reply:
A proper init function for the API would also avoid making this API return errors, and always succeed (since the shm dance is either moved to the initialization of the module, or removed altogether as discussed 1-1) - let's no go down that road yet though...